src/main/java/picard/annotation/Gene.java

```diff
@@ -64,7 +64,9 @@ public int compareTo(final Gene that) {
     }
 
     public boolean equals(final Object other) {
-        if (!(other instanceof Gene)) return false;
+        if (this == other) return true;
+        if (other == null) return false;
+        if (this.getClass() != other.getClass()) return false;
         else if (this == other) return true;
```

Comment:
useless else if since if (this == other was already checked in an above if statement.